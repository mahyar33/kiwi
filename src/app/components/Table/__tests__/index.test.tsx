import React from 'react';
import { render, screen } from '@testing-library/react';
import { Table } from '../index';

const renderWithProps = (props: Parameters<typeof Table>[number]) =>
  render(<Table {...props} />);
const tHead = [
  {
    dataField: 'id',
    text: '#',
  },
  {
    dataField: 'cityFrom',
    text: 'From',
  },
  {
    dataField: 'cityTo',
    text: 'To',
  },
  {
    dataField: 'dTime',
    text: 'Date And Time',
  },
  {
    dataField: 'price',
    text: 'Price',
  },
];

describe('<Table />', () => {
  const props: Parameters<typeof Table>[number] = {
    noResult: 'text',
    tHead,
    tBody: [
      {
        dTime: '10:30' + '2021/1/1',
        id: '1',
        cityTo: 'Tehran',
        cityFrom: 'New York',
        price: '1000',
      },
    ],
  };
  it('should render an <table> tag', () => {
    const table = renderWithProps(props);
    expect(table.container.querySelector('table')).toBeInTheDocument();
  });
  it('should match snapshot', () => {
    const table = renderWithProps(props);
    expect(table.container.firstChild).toMatchSnapshot();
  });
  it('should have 1 rows in tbody', () => {
    const table = renderWithProps(props);
    expect(table.container.querySelectorAll('tbody tr').length).toBe(1);
  });
  it('should have props displayed', () => {
    const table = renderWithProps(props);
    expect(
      table.queryByText(props.tBody[0].cityTo, { exact: false }),
    ).toBeInTheDocument();
  });
  it('should show noResult', () => {
    props.tBody = [];
    const table = renderWithProps(props);
    expect(table.queryByText(props.noResult)).toBeInTheDocument();
  });
  it('should match snapshot with noResult', () => {
    const table = renderWithProps(props);
    expect(table.container.firstChild).toMatchSnapshot();
  });
});
