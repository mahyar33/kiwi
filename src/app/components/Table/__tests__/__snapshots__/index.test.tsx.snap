// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should match snapshot 1`] = `
.c0 {
  border: 1px solid #ccc;
  color: black;
  width: 100%;
  margin: 2rem 0;
  border-collapse: collapse;
  word-break: keep-all;
}

.c0 thead {
  background-color: #cccccc;
}

.c0 th {
  vertical-align: bottom;
  padding: 0.3rem;
}

.c0 td {
  border: 1px solid #ccc;
  padding: 0.3rem;
}

<table
  class="c0"
>
  <thead>
    <tr>
      <th>
        #
      </th>
      <th>
        From
      </th>
      <th>
        To
      </th>
      <th>
        Date And Time
      </th>
      <th>
        Price
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        1
      </td>
      <td>
        New York
      </td>
      <td>
        Tehran
      </td>
      <td>
        10:302021/1/1
      </td>
      <td>
        1000
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<Table /> should match snapshot with noResult 1`] = `
.c0 {
  color: white;
  background-color: #e43c3c;
  padding: 0.75rem 1.25rem;
  margin-bottom: 1rem;
  border-radius: 0.25rem;
  font-weight: bold;
  margin: 2rem 0;
}

<div
  class="c0"
  role="alert"
>
  text
</div>
`;
